api/v1alpha1/ssp_types.go

```diff
@@ -29,13 +29,24 @@ type TemplateValidator struct {
 	Placement lifecycleapi.NodePlacement `json:"placement,omitempty"`
 }
 
+type CommonTemplates struct {
+	// Namespace is the k8s namespace where CommonTemplates should be installed
+	Namespace string `json:"namespace"`
+
+	// Version defines the common-templates release to be installed
+	Version string `json:"version"`
```

Comment:
Do we want the version to be configurable? Is it not enough to use a hard-coded constant, that would change as new versions are released?

If we want it to be configurable, a user could change it on an existing resource. There would need to be logic in `reconcileTemplates` method, that would remove old versions of templates and create new versions.

Or maybe we could disallow changing the  `Version` once the CR is created. We would probably need to create a validating webhook for that.

The same applies for the `Namespace` parameter.

Reply:
I agree with you that the version field should be dropped and only the latest version should be deployed, I think it is pointless to have it as we are not required to be downgradeable.

Reply:
And I also think that a webhook should exist to prevent changing the namespace on an existing SSP resource.
I think `operator-sdk` provides a webhook generator

Reply:
We can add the webhook in a new PR.